src/Rakaposhi.Core/Rakaposhi.Business.Core/DBRepository/RoleRepository.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Rakaposhi.Business.Core.BaseRepository;
+using Rakaposhi.Business.Core.DataObjects;
+using Rakaposhi.Data;
+using System.Data.Common;
+using System.Text.Json;
+
+namespace Rakaposhi.Business.Core.DBRepository
+{
+    public class RoleRepository : IRoleRepository
+    {
+        private IDatabase _db;
+
+        public RoleRepository()
+        {
+            _db = Database.Instance;
+        }
+
+        public void Add(Role entity)
+        {
+            DbCommand cmd = _db.CreateStoreProcedure(storeProcedureName: StoreProcds.INSERT);
+            _db.AddParameter(cmd, paramName: Params.NAME, entity.Name);
+            _db.AddParameter(cmd, paramName: Params.DESCRIPTION, entity.Description);
+            entity.RecId = Convert.ToInt64(_db.ExecuteScalar(cmd));
+        }
+
+        public void Update(Role entity)
+        {
+            DbCommand cmd = _db.CreateStoreProcedure(storeProcedureName: StoreProcds.UPDATE);
+            _db.AddParameter(cmd, paramName: Params.RECID, paramvalue: entity.RecId);
+            _db.AddParameter(cmd, paramName: Params.NAME, paramvalue: entity.Name);
+            _db.AddParameter(cmd, paramName: Params.DESCRIPTION, paramvalue: entity.Description);
+            _db.Execute(cmd);
+        }
+
+        public void Delete(Role entity)
+        {
+            DbCommand cmd = _db.CreateStoreProcedure(storeProcedureName: StoreProcds.DELETE);
+            _db.AddParameter(cmd, paramName: Params.RECID, paramvalue: entity.RecId);
+            _db.Execute(cmd);
+        }
+
+        public Role Find(long Id)
+        {
+            DbCommand cmd = _db.CreateStoreProcedure(storeProcedureName: StoreProcds.FIND);
+            _db.AddParameter(cmd, paramName: Params.RECID, paramvalue: Id);
+            string found = Convert.ToString(_db.ExecuteScalar(cmd));
+            Role role = string.IsNullOrEmpty(found) ? null: JsonSerializer.Deserialize<Role>(found);
```

Comment:
Please remove this IsNullOrEmpty check - I will fix it

Reply:
noted.

Reply:
removed.